dialogflow/api/Test/DialogflowTest.cs

```diff
@@ -72,4 +84,65 @@ public ConsoleOutput RunWithSessionId(string command, params object[] args)
             return Run(command, arguments.ToArray());
         }
     }
+
+    // TODO: Move this class into test helpers.
+    /// <summary>
+    /// Schedules throttling.
+    /// </summary>
+    class ThrottleTokenPool
```

Comment:
Should the pool be static? Can you have multiple instances of throttler pools that in fact block requests to the same actual resource?

Reply:
I see that you create a static instance of the pool earlier, but IMO the class itself should also be static and should expose methods to initialize and get access to the throttle pool per resource type that is being throttled.